### PR TITLE
Adds ability to create hanging indents

### LINKED ISSUE #769: Implement hanging indent
Maybe Related: #582

For citations we need hanging indents. Here's a screenshot of a flash widget used for this purpose in a module built in the old system:

image

### PR DESCRIPTION
**Pull request by AnthonyRodriguez726**
Added the hanging indent feature for text chunks as well as lists. Resolves #769

### CODE REVIEW

File: packages/app/obojobo-express/routes/api/events/create_caliper_event_from_client_event_req.js [hunk 1]
```diff
@@ -124,6 +124,15 @@ module.exports = req => {
...
 
+		case 'question:submitResponse':
+			return caliperEvents.createPracticeQuestionSubmittedEvent({
+				actor: actorFromType(ACTOR_USER),
+				draftId: currentDocument.draftId,
+				contentId: currentDocument.contentId,
+				questionId: clientEvent.payload.questionId,
+				sessionIds
+			})
+
```
Comment (qwertynerd97) [inline packages/app/obojobo-express/routes/api/events/create_caliper_event_from_client_event_req.js]: This code (and the related test and snapshot) looks like it's from a different pull request, and snuck its way in somehow.

File: packages/obonode/obojobo-chunks-code/components/line/editor-component.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import React from 'react'
 
 const Line = props => {
 	return (
-		<span className={'text align-left'} data-indent={props.node.data.get('content').indent}>
+		<span className={'text align-left'} data-indent={props.node.data.get('content').indent} data-hanging-indent={props.node.data.get('content').hangingIndent}>
```
Comment (qwertynerd97) [inline packages/obonode/obojobo-chunks-code/components/line/editor-component.js]: Does code need a hanging indent attribute from the content? A single line of code never wraps, it extends infinitely and creates a code scrollbar instead.

File: packages/obonode/obojobo-chunks-text/components/line/__snapshots__/editor-component.test.js.snap [hunk 3]
```diff
@@ -3,6 +3,7 @@
 exports[`Line Editor Node Line builds the expected component 1`] = `
 <span
   className="text align-0"
+  data-hanging-indent={0}
```
Comment (qwertynerd97) [inline packages/obonode/obojobo-chunks-text/components/line/__snapshots__/editor-component.test.js.snap]: This hanging-indent is a 0, but the others are either false or true.  Maybe check the values in the editor-component.test file?

File: packages/app/obojobo-document-engine/src/scripts/common/models/obo-model.js [hunk 4]
```diff
@@ -444,5 +444,5 @@ OboModel.getRoot = function() {
 }
 
 class OboModelCollection extends Backbone.Collection {}
-
+window.__s = OboModel
```
Comment (qwertynerd97) [inline packages/app/obojobo-document-engine/src/scripts/common/models/obo-model.js]: Are we using this anywhere?  If not, we might want to remove it.

**Comment by qwertynerd97** [comment]
Outside the scope of the review: keep an eye on #989 - if this PR gets merged in first, the other PR may need to be tweaked to include the toolbar changes. (And vice versa if the other PR gets merged in first)

**Comment by zachberry** [comment]
Work on this has diverged significantly, closing in favor of #1166 which took this work but rebased it on current dev/9

**Pull request closed without merging**